src/Files.App/Shell/ShellNewMenuHelper.cs

```diff
@@ -68,20 +72,23 @@ private static async Task<ShellNewEntry> GetShellNewRegistryEntries(RegistryKey
 			return null;
 		}
 
-		private static async Task<ShellNewEntry> ParseShellNewRegistryEntry(RegistryKey key, RegistryKey root)
+		private static Task<ShellNewEntry> ParseShellNewRegistryEntry(RegistryKey key, RegistryKey root)
```

Comment:
@gave92 removing the `async` results in unwrapped returns, it propagates `NullReferenceException` to the caller

we should either restore the async, or manually wrap the non-Task return values in `Task.FromResult<ShellNewEntry?>(null)`

Reply:
Done 2 hours ago :)